### ISSUE TITLE
[BUG] Invalid byte sequence in UTF-8

### ISSUE THREAD

**Issue by Youngv** [mcorino/plat4m#1]
system: Windows 11
language: chinese
ruby 3.1.6p260 (2024-05-29 revision a777087be6) [x64-mingw-ucrt]

```shell
C:\>ruby -e "p RUBY_PLATFORM"
"x64-mingw-ucrt"
```
found the issue same with https://github.com/mcorino/wxRuby3/issues/302

```shell
C:\>ver
Microsoft Windows [版本 10.0.22631.4169]
```

https://github.com/mcorino/plat4m/blob/2e80faa90edfa35bdf793d2d96bc68dd9b406969/lib/plat4m/sniff/windows.rb#L38-L39
https://github.com/mcorino/wxRuby3/blob/3e922f789cd6ac40f6533fd4930574b26bd06fce/rakelib/lib/config.rb#L559

```
irb(main):001:0> require 'plat4m'
=> true
irb(main):002:0> Plat4m.current
C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/sniff/windows.rb:39:in `match': invalid byte sequence in UTF-8 (ArgumentError)
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/sniff/windows.rb:39:in `match'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/sniff/windows.rb:39:in `get_distro'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/sniff/windows.rb:19:in `detect_system'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/system.rb:163:in `detect_current'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/plat4m-1.1.2/lib/plat4m/system.rb:145:in `current'
        from (irb):2:in `<main>'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from C:/Ruby31-x64/bin/irb:33:in `load'
        from C:/Ruby31-x64/bin/irb:33:in `<main>'
```

**Comment by Youngv** [comment]
found a workaround with the issue, run below command in cmd first, this command alters the console's code page to 437 (United States English).
```
chcp 437
```

**Comment by mcorino** [comment]
Ah, now it starts to make sense ;-)
However, I do not really like your workaround. It feels a little 'clunky' and frankly I do not like to mess around with a users system. I would definitely prefer a more 'system independent solution'.

Could you please check what the output of the following Ruby code is on your system (I do not have any system with Chinese codepage available for testing).

```ruby
s = `for /f "tokens=2 delims=[]" %s in ('ver') do @(for /f "tokens=2" %v in ('echo %s') do echo %v)`.strip
puts "Version = #{s}"
```

On my system this produces just the version number without any of the decoration text from the `ver` output.

**Comment by Youngv** [comment]
@mcorino output of the code
```
c:\>irb
irb(main):001:0> s = `for /f "tokens=2 delims=[]" %s in ('ver') do @(for /f "tokens=2" %v in ('echo %s') do echo %v)`.strip
=> "c:\\>echo 10.0.22631.4169 \n10.0.22631.4169"
irb(main):002:0> puts "Version = #{s}"
Version = c:\>echo 10.0.22631.4169
10.0.22631.4169
=> nil
irb(main):003:0> `ver`
=> "\nMicrosoft Windows [\xB0汾 10.0.22631.4169]\n"
```

**Comment by mcorino** [comment]
Ok, that's perfect (the code sample I sent you was just missing another '@' before 'echo %v' which is why the ouput was not correct yet).
I'll release an update to the gem implementing this.

**Comment by mcorino** [comment]
This should be fixed in the 1.1.3 release.
